src/components/SelectionList/BaseListItem.tsx

```diff
@@ -23,6 +24,7 @@ function BaseListItem<TItem extends ListItem>({
     onCheckboxPress,
     onDismissError = () => {},
     rightHandSideComponent,
+    checkmarkDirection = 'left',
```

Comment:
This should not be done this way... we just need new ListItem type. Please check how we've done this with UserListItem, RadioListItem and TableListItem

Reply:
We will continue with this approach and make these changes later https://expensify.slack.com/archives/C01GTK53T8Q/p1710333711044009?thread_ts=1710332452.771549&cid=C01GTK53T8Q